packages/cli/create-strapi-app/src/utils/template.ts

```diff
@@ -0,0 +1,208 @@
+import path from 'node:path';
+import url from 'node:url';
+import { Readable } from 'node:stream';
+import { pipeline } from 'node:stream/promises';
+import * as tar from 'tar';
+import retry from 'async-retry';
+
+import fse from 'fs-extra';
+import type { Scope } from '../types';
+
+const stripTrailingSlash = (str: string) => {
+  return str.endsWith('/') ? str.slice(0, -1) : str;
+};
+
+// Merge template with new project being created
+export async function copyTemplate(scope: Scope, rootPath: string) {
+  const { template } = scope;
+
+  if (!template) {
+    throw new Error('Missing template or example app option');
+  }
+
+  if (await isOfficialTemplate(template, scope.templateBranch)) {
+    await retry(
+      () =>
+        downloadGithubRepo(rootPath, {
+          owner: 'strapi',
+          repo: 'strapi',
+          branch: scope.templateBranch,
+          subPath: `templates/${template}`,
+        }),
+      {
+        retries: 3,
+        onRetry(err, attempt) {
+          console.log(`Retrying to download the template. Attempt ${attempt}. Error: ${err}`);
+        },
+      }
+    );
+
+    return;
+  }
+
+  if (isLocalTemplate(template)) {
+    const filePath = template.startsWith('file://') ? url.fileURLToPath(template) : template;
+
+    await fse.copy(filePath, rootPath);
+  }
+
+  if (isGithubShorthand(template)) {
+    const [owner, repo, ...pathSegments] = template.split('/');
+    const subPath = pathSegments.length ? pathSegments.join('/') : scope.templatePath;
+
+    await retry(
+      () => downloadGithubRepo(rootPath, { owner, repo, branch: scope.templateBranch, subPath }),
+      {
+        retries: 3,
+        onRetry(err, attempt) {
+          console.log(`Retrying to download the template. Attempt ${attempt}. Error: ${err}`);
+        },
+      }
+    );
+
+    return;
+  }
+
+  if (isGithubRepo(template)) {
+    const url = new URL(template);
+
+    const [owner, repo, t, branch, ...path] = stripTrailingSlash(url.pathname.slice(1)).split('/');
```

Comment:
Can you use a name other than `path` so it doesn't conflict with the library?

Reply:
And I noticed it because '/' isn't necessarily guaranteed in a tar if the tar was made on windows, but in this case github probably does guarantee it, so it's probably fine but worth checking.

Reply:
yeah with github it's guaranteed 👍  will rename